hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSSchedulerNode.java

```diff
@@ -110,6 +132,44 @@ synchronized FSAppAttempt getReservedAppSchedulable() {
   }
 
   /**
+   * List reserved resources after preemption and assign them to the
+   * appropriate applications in a FIFO order.
+   * @return if any resources were allocated
+   */
+  synchronized boolean assignContainersToPreemptionReservees() {
```

Comment:
I am slightly concerned about this splitting the container allocation code across too many places. The alternative is to do this in FairScheduler but maintain the map here, which is also ugly. Would it make any sense to hold this information in FSContext? 

Just calling out we need to think through this carefully.